### PR TITLE
Create Employees' categories

### PR DESCRIPTION
**Pull request by alinaioanagavrila**
I created all employees' categories.

### CODE REVIEW

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 1]
```diff
@@ -1,15 +1,143 @@
...
+import com.example.alinagavrila.employees_administration.constants.Enums.ProgrammingLanguage;
 import com.example.alinagavrila.employees_administration.constants.Enums.Gender;
 
 import java.util.ArrayList;
 import java.util.List;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+
+import javax.xml.namespace.NamespaceContext;
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: is this import used?

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 2]
```diff
@@ -1,15 +1,143 @@
...
 
 public class Company {
     private static List<Employee> employeeList = new ArrayList<>();
 
-    public static void addNewEmployee(Employee employee) {
-        employeeList.add(employee);
+    public static Employee newEmployee() throws IOException {
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: this method should be renamed. it's not very intuitive what it is supposed to do

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 3]
```diff
@@ -1,15 +1,143 @@
...
+        Employee employee = null;
+        Gender gender;
+        BufferedReader bufferReader = new BufferedReader(new InputStreamReader(System.in));
+        String type = null;
+        System.out.print("New employee type: ");
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: you can wait the input on a new line, so you can use println and avoid concatenating the comments with \n

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 4]
```diff
@@ -1,15 +1,143 @@
...
+            e.printStackTrace();
+        }
+
+        System.out.print("\nName:");
+        String name = bufferReader.readLine();
+        System.out.println("\nCNP:");
+        String CNP = bufferReader.readLine();
+        System.out.println("\nGender:");
+        String gen = bufferReader.readLine();
+        if(gen.equals("F")) {
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: use IDE's option to format the code

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 5]
```diff
@@ -1,15 +1,143 @@
...
+            gender = Gender.F;
+        }else {
+            gender = Gender.M;
+        }
+
+        switch(type) {
+            case "Administrative":
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: move all these hardcoded strings in constants package

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 6]
```diff
@@ -1,15 +1,143 @@
...
+                System.out.print("\nApproves Leave: ");
+                boolean approveLeaves = Boolean.parseBoolean(bufferReader.readLine());
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: typo "approves"

Comment (rbutnariu) [reply app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: also, what happens if the user is not providing a boolean value? (the input is not true/false, yes/no)

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 7]
```diff
@@ -1,15 +1,143 @@
...
+                System.out.print("\nIs Calculating The Salaries: ");
+                boolean isCalculatingTheSalaries = Boolean.parseBoolean(bufferReader.readLine());
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: same here

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java [hunk 8]
```diff
@@ -1,15 +1,143 @@
...
+                        break;
+                    default:
+                        System.out.println("Not a valid programming language");
+                }
+
+                if (type.equals("Dev")) {
+                    employee = new DevEmployee(name, CNP, gender, type, programmingLanguage);
+                } else {
+                    System.out.print("\nNumber Of Team Members: ");
+                    int numberOfTeamMembers = Integer.valueOf(bufferReader.readLine());
```
Comment (rbutnariu) [inline app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: same here

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java (hunk 5 above)
Comment (alinaioanagavrila) [reply app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: Done

File: app/src/main/java/com/example/alinagavrila/employees_administration/Company.java (hunk 1 above)
Comment (alinaioanagavrila) [reply app/src/main/java/com/example/alinagavrila/employees_administration/Company.java]: no

**Pull request merged**